File: eSalon/eSalon.Model/Promocija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eSalon.Model
{
    public class Promocija
    {
        public int PromocijaId { get; set; }
        public string Naziv { get; set; } = null!;
        public string? Opis { get; set; }
        public string Kod { get; set; } = null!;
        public decimal Popust { get; set; }
        public DateTime DatumPocetka { get; set; }
        public DateTime DatumKraja { get; set; }
        public int UslugaId { get; set; }
        public string UslugaNaziv { get; set; } = string.Empty;
        public byte[]? SlikaUsluge { get; set; }
        public bool? Status { get; set; }
    }
}

File: eSalon/UI/esalon_mobile/lib/models/promocija.dart
import 'package:json_annotation/json_annotation.dart';

part 'promocija.g.dart';

@JsonSerializable()
class Promocija {
  int? promocijaId;
  String? naziv;
  String? opis;
  String? kod;
  double? popust;
  DateTime? datumPocetka;
  DateTime? datumKraja;
  int? uslugaId;
  String? uslugaNaziv;
  String? slikaUsluge;
  bool? status;
  bool? jeBuduca;

  Promocija({
    this.promocijaId,
    this.naziv,
    this.opis,
    this.kod,
    this.popust,
    this.datumPocetka,
    this.datumKraja,
    this.uslugaId,
    this.uslugaNaziv,
    this.slikaUsluge,
    this.status,
    this.jeBuduca,
  });

  factory Promocija.fromJson(Map<String, dynamic> json) =>
      _$PromocijaFromJson(json);
  Map<String, dynamic> toJson() => _$PromocijaToJson(this);
}

